File: 5th-Semester/Design-Pattern/refactoring-workshop/ruby/lib/fizz_buzz.rb
class FizzBuzz
  def self.say(input)
    if input % 15 == 0
      result = 'FizzBuzz'
    else
      if input % 3 == 0
        result = 'Fizz'
      end

      if input % 5 == 0
        result = 'Buzz'
      end
    end

    return result if result
    input.to_s
  end
end

File: 5th-Semester/Design-Pattern/refactoring-workshop/js/src/fizz_buzz.js
/**
 * Requirements:
 * For factor of three print Fizz instead of the number
 * For factor of five print Buzz instead of the number
 * For numbers which are factors of both three and five print FizzBuzz instead of the number
 */
function say(number) {
  let strReturn

  if (number % 15 == 0) {
    strReturn = "FizzBuzz"
  } else {
    if (number % 3 == 0) strReturn = "Fizz"

    if (number % 5 == 0) strReturn = "Buzz"
  }

  if (strReturn != undefined) return strReturn

  return number.toString()
}

export default say